lib/net/ldap/auth_adapter/sasl.rb

```diff
@@ -47,7 +47,7 @@ def bind(auth)
             end
 
             return pdu unless pdu.result_code == Net::LDAP::ResultCodeSaslBindInProgress
-            raise Net::LDAP::SASLChallengeOverflowError, "sasl-challenge overflow" if ((n += 1) > MaxSaslChallenges)
+            raise Net::LDAP::SASLChallengeOverflowError, "sasl-challenge overflow" if ((n += 1) > 10)
```

Comment:
How do you came up with value 10? Isn't that variable "MaxSaslChallenges" used by any other place?